### PR TITLE
Reconciler Gherkin tests (II)

### PR DESCRIPTION
**Pull request by tobiscr**
**Description**

Add additional E2E test cases for reconciler.

**Related issue(s)**
https://github.com/kyma-project/kyma-operator/issues/76

### CODE REVIEW

File: tests/components/reconciler/Makefile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+ifdef DEBUG
+VERBOSE = -v
+endif
+
+.PHONY: resolve
+resolve:
+	go mod tidy -compat=1.17
```
Comment (jakobmoellersap) [inline tests/components/reconciler/Makefile:7]: This is unnecessary, you can already base the module on go1.19 without doing this

File: tests/components/reconciler/go.mod [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module github.com/kyma-project/kyma/tests/components/reconciler
+
+go 1.18
```
Comment (jakobmoellersap) [inline tests/components/reconciler/go.mod:3]:
```suggestion
go 1.19
```

**Review by jakobmoellersap — CHANGES_REQUESTED** [review]
Tests failing and go version outdated, other than that LGTM!

File: tests/components/reconciler/Makefile (hunk 1 above)
Comment (tobiscr) [reply tests/components/reconciler/Makefile]: fixed :)

**Review by a-thaler — APPROVED** [review]

**Review by jakobmoellersap — APPROVED** [review]

**Pull request merged**